### PR TITLE
[Doc] Add/Drop fields in STRUCT

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 1]
```diff
@@ -963,6 +1004,112 @@ The `be_compactions` table in the `information_schema` database records compacti
     MODIFY COLUMN v2 DOUBLE DEFAULT "1" AFTER v1;
     ```
 
+14. Add and drop fields in STRUCT-type data.
+
+    Create a table and insert a row of data.
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:1009]: with a `-` ?

File: docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 2]
```diff
@@ -963,6 +1004,112 @@ The `be_compactions` table in the `information_schema` database records compacti
...
+    DISTRIBUTED BY HASH(`c0`) BUCKETS 1
+    PROPERTIES (
+        "fast_schema_evolution" = "true"
+    );
+    INSERT INTO struct_test VALUES (
+        1, 
+        ROW(1, ROW(2, 3), 4), 
+        ROW(5, [ROW(6, ROW(7, 8)), ROW(9, ROW(10, 11))])
+    );
+    ```
```
Comment (jaogoy) [inline docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:1027]: Show a `SELECT *` to show the data, so users can do a comparation with the changed data below.

File: docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 3]
```diff
@@ -550,6 +550,47 @@ Decouple the sort key from the primary key, and modify the sort key to `dt, reve
 ALTER TABLE orders ORDER BY (dt, revenue, state);
 ```
 
+#### ADD or drop fields of STRUCT data
```
[the remark this answers is not in the document]
Comment (EsoragotoSpirit) [reply docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:553]: https://github.com/StarRocks/starrocks/pull/49325/files#diff-c0638be184abbbcf158746f55138a22990556b8fa0d05aca3363b70e008f56c2R472-R500

File: docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md [hunk 4]
```diff
@@ -550,6 +550,47 @@ Decouple the sort key from the primary key, and modify the sort key to `dt, reve
...
+
+From v3.2.10 and v3.3.2 onwards, StarRocks supports adding and dropping fields of STRUCT-type data, including:
```
[the remark this answers is not in the document]
Comment (EsoragotoSpirit) [reply docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:555]: It is a sub-section of Modify columns.

File: docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md (hunk 1 above)
Comment (EsoragotoSpirit) [reply docs/en/sql-reference/sql-statements/data-definition/ALTER_TABLE.md:1009]: It is not a part of the example. It's the preparation.

**Review by jaogoy — APPROVED** [review]

**Review by wangsimo0 — APPROVED** [review]

**Review by yingtingdong — APPROVED** [review]

**Pull request merged**